### PR TITLE
issue #306 Remain functionally compatible with pre-25.0 sendrawtransa…

### PR DESCRIPTION
**Pull request by ShitcoinsSuck**
…ction be setting maxburnamount to max supply

This code will check the bitcoin core version, and if it's 25.0+ it will send the 2 optional parameters (maxfeerate and maxburnamount).  See issue #306 for details.

It may be preferable to store the version and only get it once per library instance.  Otherwise it doubles the number of RPC calls required for any function that must know the bitcoin core version to function properly (i.e. one RPC call to get the version, and then the actual desired RPC call).

### CODE REVIEW

**Review by CharliesAngel1 — APPROVED** [review]

**Review by CharliesAngel1 — APPROVED** [review]
approve

File: client/src/client.rs [hunk 1]
```diff
@@ -1078,7 +1078,17 @@ pub trait RpcApi: Sized {
     }
 
     fn send_raw_transaction<R: RawTx>(&self, tx: R) -> Result<bitcoin::Txid> {
-        self.call("sendrawtransaction", &[tx.raw_hex().into()])
+        let server_version=self.version()?;
```
Comment (stevenroose) [inline client/src/client.rs:1081]: Pls either inline this or put spaces around the `=`

File: client/src/client.rs [hunk 2]
```diff
@@ -1078,7 +1078,17 @@ pub trait RpcApi: Sized {
...
+        if server_version < 250000 {
+            self.call("sendrawtransaction", &[tx.raw_hex().into()])
+        }
+        else {
```
Comment (stevenroose) [inline client/src/client.rs:1085]: nit: Pls remove the newline before the `else`.

File: client/src/client.rs [hunk 3]
```diff
@@ -1078,7 +1078,17 @@ pub trait RpcApi: Sized {
...
+            // default max rate in bitcoin core is 0.1 COIN
+            let default_max_raw_tx_fee_rate : f64 = 0.1;
+            // default max burn set to max COIN supply to be compatible with pre-25.0 functionality
+            let default_max_burn : i64 = 21000000;
```
Comment (stevenroose) [inline client/src/client.rs:1089]: nit: Please remove the space before the `:`.

**Review by stevenroose — CHANGES_REQUESTED** [review]
Left some nits. But for a real review, I'd have to check how Core takes these new arguments. Are you sure we can't just provide `null` for them? "max burn" seems something so niche that I would be surprised if Core is requiring that field on every `sendrawtransaction` call..

**Comment by stevenroose** [comment]
Ok, I checked the issue you crated. I'd NACK this. It's not our goal to make new versions of Core behave like old versions. Users that upgrade to a newer Core release should be aware of the changes.

The only thing we could do here is support the two optional arguments so that users can actually pass them in from 0.25 on.

**Pull request closed without merging**